README.md

```diff
@@ -101,7 +101,7 @@ You can then proceed to run [our examples](https://github.com/pytorch/data/tree/
 ### From source
 
 ```bash
-python setup.py install
+pip install .
```

Comment:
Question:
1. Just to confirm, this is recommended for all users (not just MacOS)?
2. I wonder if PyTorch Core is impacted as well?

Reply:
Curious why replace `python setup.py install` with `pip install .` solve the issue? (maybe it's explained somewhere but didn't see it 😳)

Reply:
> 1. Just to confirm, this is recommended for all users (not just MacOS)?

Yea, it's a recommendation from Python

> 2. I wonder if PyTorch Core is impacted as well?

IDK. As far as I know, it doesn't

> Curious why replace `python setup.py install` with `pip install .` solve the issue?

The issue itself is mysterious enough to me as when doing `python setup.py install` it added `torchdata` as its dependency -> circular dependency. See log: https://github.com/pytorch/data/actions/runs/3972365525/jobs/6810200117#step:9:1639
I don't know why it happens only on macos python 3.9 and python 3.10.

So, I have no clue why `pip install` can solve this problem. Since it does solve it anyway, I decide not to spend extra time to dive deeper on this topic until I have more time.